notebooks/evidence_matching/fusion_evidence_matching.ipynb

```diff
@@ -0,0 +1,396 @@
+    "We expect matches for the first fusion as its partners are both KIF5B and RET (and have\n",
+    "equivalent breakpoint locations), and the third fusion, as its 5' partner is a multiple \n",
+    "possible genes element and its 3' partner is RET."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "483fb8cc",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:cool_seq_tool.mappers.exon_genomic_coords:48406078 on NC_000023.11 does not occur within the exons for NM_005636.4\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Generate CategoricalFusion objects from CIViC\n",
+    "from fusor.harvester import CIVICHarvester\n",
+    "\n",
+    "harvester = CIVICHarvester(fusions_list=variants)\n",
+    "fusions_list = harvester.load_records()\n",
+    "\n",
+    "categorical_fusions = []\n",
+    "for fusion in fusions_list:\n",
+    "    if \"?\" in fusion.vicc_compliant_name:\n",
+    "        continue\n",
+    "    cex = await translator.from_civic(civic=fusion)\n",
+    "    categorical_fusions.append(cex)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "52333fb7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define evidence matching method\n",
+    "from fusor.models import (\n",
+    "    AssayedFusion,\n",
+    "    CategoricalFusion,\n",
+    "    GeneElement,\n",
+    "    MultiplePossibleGenesElement,\n",
+    "    TranscriptSegmentElement,\n",
+    "    UnknownGeneElement,\n",
+    ")\n",
+    "\n",
+    "\n",
+    "def compare_structure(assayed_element: TranscriptSegmentElement | UnknownGeneElement | GeneElement,\n",
+    "                      categorical_element: TranscriptSegmentElement | MultiplePossibleGenesElement |\n",
+    "                      GeneElement, is_five_prime_partner: bool) -> tuple[bool | str, int]:\n",
+    "    \"\"\"Compare transcript segments for an assayed and categorical fusions\n",
+    "    :param assayed_transcript: The assayed fusion transcript or unknown gene element or gene element\n",
+    "    :param categorical_transcript: The categorical fusion transcript or mulitple possible genes element\n",
+    "    :param is_five_prime_partner: If the 5' fusion partner is being compared\n",
+    "    :return A boolean or string indicating if a match is found and a score indiciating the degree of match\n",
+    "    \"\"\"\n",
+    "    # Set default match score\n",
+    "    match_score = 0\n",
+    "\n",
+    "    # If the assayed partner is unknown or the categorical partner is a multiple possible gene element, return match score of 0 as no precise information\n",
+    "    # regarding the compared elements is known\n",
+    "    if isinstance(assayed_element, UnknownGeneElement) or isinstance(categorical_element, MultiplePossibleGenesElement):\n",
+    "        return \"NA\", 0\n",
```

Comment:
Your return type states boolean, but this is returning a string

Reply:
I think the return type allows for a string too? The function lists:
`tuple[bool | str, int]`

Reply:
Oh, that's my bad. I thought it said `tuple[bool, str | int]`